app/Entirety/entirety/widgets.py

```diff
@@ -15,3 +15,38 @@ def __init__(self, choices=[], attrs=None):
 
     def decompress(self, value):
         return ["", ""]
+
+
+class DropdownOrTextWidget(forms.MultiWidget):
+    def __init__(self, choices=[], **kwargs):
+        attrs = {}
+        pattern = None
+        if "selector" in kwargs and kwargs["selector"] != None:
+            if kwargs["selector"] == "pattern":
+                if kwargs["pattern"] != None:
+                    attrs["hidden"] = "hidden"
+                    pattern = kwargs["pattern"]
+            elif kwargs["selector"] == "id":
+                attrs["initial"] = kwargs["id"]
+            else:
+                logger.error(f"Do not understand selector: {kwargs['selector']}")
+
+        self.widgets = (
+            forms.Select(
+                choices=choices,
+                attrs=attrs,
+            ),
+            forms.TextInput(
+                attrs={
+                    "data-bs-toggle": "tooltip",
+                    "data-bs-placement": "top",
+                    "title": "Entity type or type pattern.",
```

Comment:
This tooltip title appears both in entity type and entity id. May be title can also be passed as a parameter.

Reply:
done.